src/Entity/ContentItem.php

```diff
@@ -35,22 +35,33 @@ public function getShortType($infix = '-')
      *
      * @param ContentItemInterface $from
      * @param ContentItemInterface $to
-     * @return ContentItemInterface
+     * @return array
      */
     public static function convert(ContentItemInterface $from, ContentItemInterface $to)
     {
-        $reflectionFrom = new \ReflectionClass($from);
-        $reflectionTo = new \ReflectionClass($to);
-
-        foreach ($reflectionFrom->getProperties() as $property) {
-            $property->setAccessible(true);
-            $method = 'set' . ucfirst($property->getName());
-            if ($reflectionTo->hasMethod($method)) {
-                $to->$method($property->getValue($from));
-            }
+        $properties = [];
+        try {
+            $reflectionFrom = new \ReflectionClass($from);
+            $reflectionTo = new \ReflectionClass($to);
+
+            do {
+                $convertToType = [];
+                /* @var $p \ReflectionProperty */
+                foreach ($reflectionFrom->getProperties() as $property) {
+                    $property->setAccessible(true);
+                    $method = 'set' . ucfirst($property->getName());
+
+                    if ($reflectionTo->hasMethod($method)) {
+                        $to->$method($property->getValue($from));
+                        $convertToType[$method] = $to->$method($property->getValue($from));
+                    }
+                }
+                $properties = array_merge($convertToType, $properties);
+            } while ($reflectionFrom = $reflectionFrom->getParentClass());
+        } catch (\ReflectionException $e) {
         }
```

Comment:
Misschien nog even een comment toevoegen in het lege "catch" blok waarom je een error afvangt en er niks mee hoeft te worden gedaan

Reply:
Aangepast